addon/components/item-picker/component.js

```diff
@@ -266,23 +274,32 @@ export default Component.extend({
     /**
      * When an item is clicked in the list
      */
-    onItemClick (item) {
+    onItemClick (model) {
       if (this.get('selectMultiple')) {
+        let validator = this.get('onSelectionValidator');
         const itemsToAdd = this.get('itemsToAdd');
-        const existingObj = itemsToAdd.findBy('id', item.id);
+        const existingObj = itemsToAdd.findBy('id', model.item.id);
         if (!existingObj) {
-          itemsToAdd.pushObject(item);
+          itemsToAdd.pushObject(model.item);
         } else {
           itemsToAdd.removeObject(existingObj);
         }
+
+        if (validator) {
+          validator(model).then(response => {
+            if (response.state !== 'ok') {
+              this.showMultiValidationMessage(response.message);
+            }
+          });
+        }
       } else {
-        if (this.get('currentItem.id') === item.id) {
+        if (this.get('currentItem.id') === model.item.id) {
           this.set('currentItem', null);
         } else {
           this.setProperties({
             errorHash: null,
             selectAnyway: false,
-            currentItem: item
+            currentItem: model.item
```

Comment:
I think the `currentItem` is what causes the Preview to be loaded... and I think we want to send that a `model`, so maybe rename `currentItem` to `currentModel`, and pass that forward to the preview. That allows the custom-or-default preview component to show any of the `model.properties` that the row has added. In the `onItemSelected` in the Preview component, at that point we return the `item` vs the `model`